certification/internal/shell/imagesource_registry.go

```diff
@@ -0,0 +1,78 @@
+package shell
+
+import (
+	"bytes"
+	"strings"
+
+	cmdchain "github.com/rainu/go-command-chain"
+	"github.com/redhat-openshift-ecosystem/openshift-preflight/certification"
+	log "github.com/sirupsen/logrus"
+)
+
+type ImageSourceRegistryCheck struct {
+}
+
+var approvedRegistries = map[string]string{
+	"registry.connect.dev.redhat.com":   "registry.connect.dev.redhat.com",
+	"registry.connect.qa.redhat.com":    "registry.connect.qa.redhat.com",
+	"registry.connect.stage.redhat.com": "registry.connect.stage.redhat.com",
+	"registry.connect.redhat.com":       "registry.connect.redhat.com",
+	"registry.redhat.io":                "registry.redhat.io",
+	"registry.access.redhat.com":        "registry.access.redhat.com",
+}
+
+func (p *ImageSourceRegistryCheck) Validate(bundleImage string) (bool, error) {
+
+	output := &bytes.Buffer{}
+
+	err := cmdchain.Builder().
+		WithInput(strings.NewReader(bundleImage)).
+		Join("cut", "-d", ",", "-f1").
+		Join("cut", "-d", "/", "-f1").
+		Finalize().WithOutput(output).Run()
+	if err != nil {
+		log.Error(" Failed to execute cmdchain builder")
```

Comment:
Would it be easier for users if we said the "could not extract registry" or "the bundle image input could not be parsed"?
looks like cmdchain builder only has one contributor so it might be a little obscure. I think if debugging is on the error message will help engineers/devs figure things out

Reply:
I'm also left unclear what exactly this bit of code is doing. It's too opaque for someone in the future to be able to change this without a comment explaining what is going on.